### DISCUSSION TITLE
Other Wallets and Chains

### DISCUSSION THREAD

**Discussion by Dilosen** [WalletConnect/WalletConnectUnity#133] [Q&A] [answered]
Hi - I would like to add Algorand specifically to this modal setup, but a general guidance would be appreciated.

I found the following [namespaces for Algorand according to CAIP 2](https://github.com/ChainAgnostic/namespaces/blob/main/algorand/caip2.md)

Algorand MainNet has the genesis hash wGHE2Pwdvd7S12BL5FaOP20EGYesN73ktiC1qzkkit8=
algorand:wGHE2Pwdvd7S12BL5FaOP20EGYesN73k

Algorand TestNet has the genesis hash SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=
algorand:SGO1GKSzyE7IEPItTxCByw9x8FmnrCDe

Algorand BetaNet has the genesis hash mFgazF+2uRS1tMiL9dsj01hJGySEmPN28B/TjjvpVW0=
algorand:mFgazF-2uRS1tMiL9dsj01hJGySEmPN2

In comparision to:

public static readonly string EvmNamespace = "eip155";

        public static readonly Chain Ethereum =
            new Chain(EvmNamespace, "1", nameof(Ethereum), new Color(0.38f, 0.49f, 0.92f));

I would assume the following:

```new Chain(algorand:SGO1GKSzyE7IEPItTxCByw9x8FmnrCDe, "1, nameof(Algorand), new Color(0f, 0f, 0f));  //Testnet```

I also noticed a UnityWebRequestWalletsFactory() that GET data from the api  - where can I define what wallets can be used? I don't seem to see it in the WalletConnect Dashboard, but it seems like we're GET that data from the api? 

Maybe I missed it somewhere?

**Comment by skibitsky** [comment]
Hi @moosesnWoop,

Almost correct!

The namespace is `algorand`.
The chain ID is the value after `:`, which is `SGO1GKSzyE7IEPItTxCByw9x8FmnrCDe` for the testnet.

The chain object would like something like this:
```csharp
Chain AlgorandTestnet = new("algorand", "SGO1GKSzyE7IEPItTxCByw9x8FmnrCDe", "Algorand Testnet", new Color(0.21f, 0.82f, 0.5f), true);
```
You will also need to modify [this line](https://github.com/WalletConnect/WalletConnectUnity/blob/main/Packages/com.walletconnect.modal/Samples~/Modal%20Sample/Scripts/Dapp.cs#L106) to include non EVM chains.

I haven't had a chance to test Algorand myself; please let me know if this approach works.

---
The `UnityWebRequestWalletsFactory` consumes data from the [WalletConnect explorer](https://walletconnect.com/explorer).

In the [WalletConnectModalOptions](https://github.com/WalletConnect/WalletConnectUnity/blob/main/Packages/com.walletconnect.modal/Runtime/WalletConnectModalOptions.cs#L9-L11), you can specify which wallets to include or exclude. You can get wallet IDs from the [WalletConnect explorer](https://walletconnect.com/explorer).

For example:
```csharp
var options = new WalletConnectModalOptions                                            
{                                                                                      
    ConnectOptions = BuildConnectOptions(),                                            
    ExcludedWalletIds = new[]                                                          
    {                                                                                  
        "c57ca95b47569778a828d19178114f4db188b89b763c899ba0be274e97267d96", // MetaMask
        "4622a2b2d6af1c9844944291e5e7351a6aa24cd7b23099efac1b2fd875da31a0", // Trust   
    }                                                                                  
};                                                                                     
                                                                                       
WalletConnectModal.Open(options);

**Reply by Dilosen** [reply]
Hello, just to update;

i've realised I'm okay with the mulitple wallet access, users should be able to chose how they want to interact, irrespective of wallet.

I'm now more focused on intergrating Algorand into the exisiting Modal.  So far, here is what I have done:

In Chain.cs:

``` 
public static readonly string AlgoNameSpace = "algorand";

        public static readonly Chain Algorand =
            new Chain(AlgoNameSpace, "SGO1GKSzyE7IEPItTxCByw9x8FmnrCDe", "Algorand Test Net", new Color(0.21f, 0.82f, 0.5f), true)
```

 
I will add mainnet and betanet after testing this. If we go to the Dapp.cs:

I've added an elseif() statement to incorporate the _selectedChains and the required namespaces. Still working on this section though

**Comment by Dilosen** [comment]
Hi 

Since I've added this, it seems to work fine, but it seems that there is no supported wallet or chain? 

I'm kinda confused at this. I thought that Algorand was supported on Wallet Connect and that there were wallets already attached?

I tried using Ledger to scan the QR code and it says it only supported X number of chains.. so then I tried metamask and only had Ethereum. There is no Pera wallet? 

I guess Eth works fine, but the chain that I want to use is Algorand- any methods to set that up?

**Reply by skibitsky** [reply, marked as answer]
Hi,

There are two ways to connect a wallet: QR code and deep link. 

Pera should work with the QR code. Please take a look at [this issue](https://github.com/WalletConnect/WalletConnectUnity/issues/115). 

However, Pera will not appear in the list of wallets and therefore will not be able to use the deep link. This is because the wallet data is pulled from the [WalletConnect Explorer](https://walletconnect.com/explorer), where the Pera wallet is absent. If the developers of Pera add it to the explorer, it will automatically appear in the Modal.

**Comment by Dilosen** [comment]
Thanks, I didn't know Pera wasn't on the WalletConnect explorer. I will try the method via QR code, which is fine. I also looked at the issue, and it seems that I have already updated the methods/events for the Algo use case, but will test it properly when I get back to my desk.

**Comment by Dilosen** [comment]
Update: Got it to connect! I also included some event/methods namely:

```
var methods = new[]
                    {
                       "algo_signTxn", 
                    };
                var events = new[]
                    {
                        "accountChanged",
                    };
```

and then created a new namespace for the Algo associated chains. 

I will look into the exact methods, but I believe that's all that's needed. I am also assuming that this way of implementation will be able to expand on any number of chains, given the CAIP naming conventions.
